### PR TITLE
[stable] Relax some new std.complex unittests for double-precision real

### PR DESCRIPTION
**Pull request by kinke**

### CODE REVIEW

**Comment by kinke** [comment]
Pinging @ibuclaw. These adaptations almost suffice to pass the unittests with LDC on Windows (at least the debug ones) with 64-bit real. I assume you did test such targets with gdc though...

File: std/complex.d [hunk 1]
```diff
@@ -1397,22 +1405,22 @@ Complex!T log(T)(Complex!T x) @safe pure nothrow @nogc
...
-    assert(isClose(d, complex(0.0L, 2.09439510239319549230L), 0.0, 1e-18));
+    assert(isClose(d, complex(0.0L, 2.09439510239319549230L), 0.0, 1e-15));
 
     auto e = log(fromPolar(1.0, 5.0 * PI / 6.0));
-    assert(isClose(e, complex(0.0L, 2.61799387799149436538L), 0.0, 1e-18));
+    assert(isClose(e, complex(0.0L, 2.61799387799149436538L), 0.0, 1e-15));
 
     auto f = log(fromPolar(1.0, PI));
-    assert(isClose(f, complex(0.0L, -3.1415926535897932384L), 0.0, 1e-18));
+    assert(isClose(f, complex(0.0L, -3.1415926535897932384L), 0.0, 1e-15));
```
Comment (kinke) [inline std/complex.d:1423]: The actual imaginary part is *positive* PI here on my box.

File: std/complex.d [hunk 2]
```diff
@@ -1441,31 +1449,31 @@ Complex!T log10(T)(Complex!T x) @safe pure nothrow @nogc
...
-    assert(isClose(d, complex(0.0L, 0.909584235894560898323L), 0.0, 1e-18));
+    assert(isClose(d, complex(0.0L, 0.909584235894560898323L), 0.0, 1e-15));
 
     auto e = log10(fromPolar(1.0, 5.0 * PI / 6.0));
-    assert(isClose(e, complex(0.0L, 1.13698029486820112290L), 0.0, 1e-18));
+    assert(isClose(e, complex(0.0L, 1.13698029486820112290L), 0.0, 1e-15));
 
     auto f = log10(fromPolar(1.0, PI));
-    assert(isClose(f, complex(0.0L, -1.36437635384184134748L), 0.0, 1e-18));
+    assert(isClose(f, complex(0.0L, -1.36437635384184134748L), 0.0, 1e-15));
```
Comment (kinke) [inline std/complex.d:1476]: *Positive* imaginary part again.

**Comment by kinke** [comment]
The sign issue seems to boil down to:
```D
void main()
{
    import std.math, core.stdc.stdio;
    const rx = sin(PI);
    const dx = sin(cast(double) PI);
    printf("%La\n", rx);
    printf("%a\n", dx);
}
```
=>
```
-0x8p-67
0x1.1a6p-53
```

Same story with `core.stdc.tgmath.sin`.

**Comment by kinke** [comment]
Exactly the same behavior/output for [C++20](https://cpp.godbolt.org/z/QKEWw8):
```C++
#include <cmath>
#include <numbers>
#include <cstdio>

int main()
{
    auto rx = std::sin(std::numbers::pi_v<long double>);
    auto dx = std::sin(std::numbers::pi_v<double>);
    printf("%La\n", rx);
    printf("%a\n", dx);
    return 0;
}
```

**Comment by ibuclaw** [comment]
> Pinging @ibuclaw. These adaptations almost suffice to pass the unittests with LDC on Windows with 64-bit real. I assume you did test such targets with gdc though...

Not on Win64 specifically, but some rudimentary tests on ARM. The tests were originally tighter, but many got loosened to accommodate for dmd CI failures.

**Comment by ibuclaw** [comment]
> Exactly the same behavior/output for [C++20](https://cpp.godbolt.org/z/QKEWw8):

To be sure, disable built-ins from being used in libstdc++, I did a lot of comparisons between the two (with built-ins turned on, I found that you get more accurate results, and proper error handling).

**Comment by kinke** [comment]
I've replaced the questionable `fromPolar(1.0, PI)` by `complex(-1.0L, 0.0L)`.

**Comment by kinke** [comment]
> To be sure, disable built-ins from being used in libstdc++

If you look at the (linked) godbolt asm, the constants are 'CTFE' computed, no sines are computed at runtime.

**Comment by ibuclaw** [comment]
> The tests were originally tighter, but many got loosened to accommodate for dmd CI failures.

Oh! And pretty much all literals used in the tests were pulled (painstakingly) from computing them on wolfram.

**Comment by thewilsonator** [comment]
Is there any reason to wait?

**Comment by ibuclaw** [comment]
> Is there any reason to wait?

Waiting for a second review.

**Pull request merged**